docs/guides/unity-catalog.md

```diff
@@ -0,0 +1,530 @@
+  member_id = databricks_user.unity_users[each.value].id
+}
+
+resource "databricks_user_role" "my_user_account_admin" {
+  provider = databricks.mws
+  for_each = toset(var.databricks_unity_admins)
+  user_id  = databricks_user.unity_users[each.value].id
+  role     = "account_admin"
+}
+```
+## Create a Unity Catalog metastore and link it to workspaces
+
+A [databricks_metastore](../resources/metastore.md) is the top level container for data in Unity Catalog. A single metastore can be shared across Databricks workspaces, and each linked workspace has a consistent view of the data and a single set of access policies. It is only recommended to have multiple metastores when organizations wish to have hard isolation boundaries between data (note that data cannot be easily joined/queried across metastores).
+
+```hcl
+resource "databricks_metastore" "this" {
+  provider      = databricks.workspace
+  name          = "primary"
+  storage_root  = "s3://${aws_s3_bucket.metastore.id}/metastore"
+  owner         = var.unity_admin_group
+  force_destroy = true
+}
+
+resource "databricks_metastore_data_access" "this" {
+  provider     = databricks.workspace
+  metastore_id = databricks_metastore.this.id
+  name         = aws_iam_role.metastore_data_access.name
+  aws_iam_role {
+    role_arn = aws_iam_role.metastore_data_access.arn
+  }
+  is_default = true
+}
+
+resource "databricks_metastore_assignment" "default_metastore" {
+  provider             = databricks.workspace
+  for_each             = toset(var.databricks_workspace_ids)
+  workspace_id         = each.key
+  metastore_id         = databricks_metastore.unity.id
+  default_catalog_name = "hive_metastore"
+}
+```
+
+## Create Unity Catalog objects in the metastore
+
+Each metastore exposes a 3-level namespace (catalog-schema-table) by which data can be organized. 
+
+```hcl
+resource "databricks_catalog" "sandbox" {
+  provider     = databricks.workspace
+  metastore_id = databricks_metastore.this.id
+  name         = "sandbox"
+  comment      = "this catalog is managed by terraform"
+  properties = {
+    purpose = "testing"
+  }
+  depends_on = [databricks_metastore_assignment.default_metastore]
+}
+
+resource "databricks_grants" "sandbox" {
+  provider = databricks.workspace
+  catalog  = databricks_catalog.sandbox.name
+  grant {
+    principal  = "Data Scientists"
+    privileges = ["USAGE", "CREATE"]
+  }
+  grant {
+    principal  = "Data Engineers"
+    privileges = ["USAGE"]
+  }
+}
+
+resource "databricks_schema" "things" {
+  provider     = databricks.workspace
+  catalog_name = databricks_catalog.sandbox.id
+  name         = "things"
+  comment      = "this database is managed by terraform"
+  properties = {
+    kind = "various"
+  }
+}
+
+resource "databricks_grants" "things" {
+  provider = databricks.workspace
+  schema   = databricks_schema.things.id
+  grant {
+    principal  = "Data Engineers"
+    privileges = ["USAGE"]
+  }
+}
+```
+
```

Comment:
How about creating managed tables?

Reply:
We'll add examples for that later. I'm not fully sure yet if we should manage tables with TF - needs more discussions.